src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -0,0 +1,478 @@
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders medium filled segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="filled" size="medium">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('pressIn', () => {
+    it('outlined segment control unselected', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('outlined segment control selected', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-1');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('filled segment control unselected', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="filled" size="small">
```

Comment:
same unselected should be without `defaultValue`?

Reply:
This is testing pressIn behaviour for unselected segment. It has a different underlay color that a selected segment